packages/vm/tests/config.js

```diff
@@ -148,7 +150,131 @@ function getRequiredForkConfigAlias(forkConfig) {
   }
   return forkConfig
 }
+
+const normalHardforks = [
+  'chainstart',
+  'homestead',
+  'tangerineWhistle',
+  'spuriousDragon',
+  'byzantium',
+  'constantinople',
+  'petersburg',
+  'istanbul',
+  'muirGlacier',
+  'berlin',
+]
+
+const transitionNetworks = {
+  ByzantiumToConstantinopleFixAt5: {
+    byzantium: 0, 
+    constantinople: 5,
+    petersburg: 5,
+    dao: null,
+    finalSupportedFork: 'petersburg',
+    startFork: 'byzantium'
+  },
+  EIP158ToByzantiumAt5: {
+    spuriousDragon: 0,
+    byzantium: 5,
+    dao: null,
+    finalSupportedFork: 'byzantium',
+    startFork: 'spuriousDragon'
+  },
+  FrontierToHomesteadAt5: {
+    frontier: 0,
+    homestead: 5,
+    dao: null,
+    finalSupportedFork: 'homestead',
+    startFork: 'frontier'
+  },
+  HomesteadToDaoAt5: {
+    homestead: 0,
+    dao: 5,
+    finalSupportedFork: 'dao',
+    startFork: 'homestead'
+  },
+  HomesteadToEIP150At5 : {
+    homestead: 0,
+    tangerineWhistle: 5,
+    dao: null,
+    finalSupportedFork: 'tangerineWhistle',
+    startFork: 'homestead'
+  }
+}
   
+/**
+ * Returns a Common for the given network (a test parameter)
+ * @param {String} network - the network field of a test
+ * @returns {Common} the Common which should be used
+ */
+function getCommon(network) {
+  const networkLowercase = network.toLowerCase()
+  if (normalHardforks.map((str) => str.toLowerCase()).includes(networkLowercase)) {
+    // normal hard fork, return the common with this hard fork
+    // find the right upper/lowercased version
+    const hfName = normalHardforks.reduce((previousValue, currentValue) => (currentValue.toLowerCase() == networkLowercase) ? currentValue : previousValue)
+    const mainnetCommon = new Common('mainnet', hfName)
+    const hardforks = mainnetCommon.hardforks()
+    const testHardforks = []
+    for (const hf of hardforks) {
+      // check if we enable this hf 
+      // disable dao hf by default (if enabled at block 0 forces the first 10 blocks to have dao-hard-fork in extraData of block header)
+      if (mainnetCommon.gteHardfork(hf.name) && hf.name != "dao") {
+        // this hardfork should be activated at block 0
+        testHardforks.push({
+          name: hf.name,
+          forkHash: hf.forkHash,
+          block: 0
+        })
+      } else {
+        // disable hardforks newer than the test hardfork (but do add "support" for it, it just never gets activated)
+        testHardforks.push({
+          name: hf.name,
+          forkHash: hf.forkHash,
+          block: null
+        })
```

Comment:
TBH I can not follow what this code actually does respectively why it is necessary to rebuild the whole hardfork stack in such a laborious manner when at the end the test is executed on the hardfork set with `const mainnetCommon = new Common('mainnet', hfName)` anyhow?

Reply:
I semi agree that this is a lot of code which does not do a lot. But if we use `mainnetCommon = new Common('mainnet', hfName)`, this means that we are (for instance) running Istanbul on block 0 while Common actually says that we should run Frontier. I do not think that is right, this is to semi future-proof the tests.

Reply:
I think you might be hanging a bit on this block-number thing too much 🙂 , Common just has this information on the underlying block numbers but is not enforcing it in any ways. And Common is used in non-block-number-dependent contexts all the time, e.g. also the VM needs to keep this as one version of using it by e.g. people expecting a Byzantium VM and then feeding it with test blocks where the number plays no role at all (and it should still  behave as Byzantium).

Anyhow, I have the impression we generally need to give this some more reflection, where and how a HF is reset and who is taking responsibility here, especially when moving over to client development.

For now we might want to keep this code, eventually we can remove later depending on the design decision we take along these questions on the VM.